src/components-examples/BUILD.bazel

```
@@ -0,0 +1,132 @@
+
+load("//tools:defaults.bzl", "ng_module", "ng_package")
+load("//tools/highlight-files:index.bzl", "highlight_files")
+load("//tools/package-docs-content:index.bzl", "package_docs_content")
+
+EXAMPLE_PACKAGES = [
+    # TODO(devversion): try to have for each entry-point a bazel package so that
+    # we can automate this using the "package.bzl" variables. Currently generated
+    # with "bazel query 'kind("ng_module", //src/components-examples/...:*)' --output="label"
+    "//src/components-examples/material/tree",
+    "//src/components-examples/material/tooltip",
+    "//src/components-examples/material/toolbar",
+    "//src/components-examples/material/tabs",
+    "//src/components-examples/material/table",
+    "//src/components-examples/material/stepper",
+    "//src/components-examples/material/sort",
+    "//src/components-examples/material/snack-bar",
+    "//src/components-examples/material/slider",
+    "//src/components-examples/material/slide-toggle",
+    "//src/components-examples/material/sidenav",
+    "//src/components-examples/material/select",
+    "//src/components-examples/material/radio",
+    "//src/components-examples/material/progress-spinner",
+    "//src/components-examples/material/progress-bar",
+    "//src/components-examples/material/paginator",
+    "//src/components-examples/material/menu",
+    "//src/components-examples/material/list",
+    "//src/components-examples/material/input",
+    "//src/components-examples/material/icon",
+    "//src/components-examples/material/grid-list",
+    "//src/components-examples/material/form-field",
+    "//src/components-examples/material/expansion",
+    "//src/components-examples/material/divider",
+    "//src/components-examples/material/dialog",
+    "//src/components-examples/material/datepicker",
+    "//src/components-examples/material/core",
+    "//src/components-examples/material/chips",
+    "//src/components-examples/material/checkbox",
+    "//src/components-examples/material/card",
+    "//src/components-examples/material/button-toggle",
+    "//src/components-examples/material/button",
+    "//src/components-examples/material/bottom-sheet",
+    "//src/components-examples/material/badge",
+    "//src/components-examples/material/autocomplete",
+    "//src/components-examples/material-experimental/popover-edit",
+    "//src/components-examples/cdk/tree",
+    "//src/components-examples/cdk/text-field",
+    "//src/components-examples/cdk/table",
+    "//src/components-examples/cdk/stepper",
+    "//src/components-examples/cdk/scrolling",
+    "//src/components-examples/cdk/portal",
+    "//src/components-examples/cdk/platform",
+    "//src/components-examples/cdk/drag-drop",
+    "//src/components-examples/cdk/clipboard",
+    "//src/components-examples/cdk/a11y",
+    "//src/components-examples/cdk-experimental/popover-edit",
+]
+
+ng_module(
+    name = "components-examples",
+    srcs = glob(["**/*.ts"]) + [":example-module.ts"],
```

Doesn't including the `glob` here just duplicate all of the deps?  We should only have to rely on `example-module.ts`

Additionally, I don't think we need to have the `:` for the `example-module.ts` `src`

```suggestion
    srcs = glob(["*.ts"]) + [":example-module.ts"],
```

or 

```suggestion
    srcs = [
      "index.ts",
      "example-data.ts",
      ":example-module.ts",
    ],
```

---

The `glob` matches only the top-level source files since all the sub folders are individual Bazel packages. We could get rid of the glob though and just fine-grain the files manually though (it's inconsistent with what we do in other modules)

And yes, the colon is not really necessary. It's valid and I use it to imply that this is a file generated by a target. Happy to change if you want me to.

---

I realized that the `:example-module.ts` was referencing a `genrule` and forgot to remove that comment.  It definitely makes sense to have it there.  Sorry.

Huh, TIL that `glob`s don't pass package boundaries, though I supposed I should has realized since if it did then this would result in multiple targets sourcing the same files. 

I think it might make sense to use the first suggestion still, `srcs = glob(["*.ts"]) + [":example-module.ts"],` since it allows for us to still use the glob, but makes it obvious that the files we are looking to include in `srcs` are just at this level.

I am fine with however you want to proceed.

---

Switched it to the top-level glob since I agree that it's more clear that way 😄

---

And yes. globs don't pass package boundaries. _Though_ I wish there would be an option to `glob` for packages! That way we could get rid of the hard-coded list of sub-example packages